examples/cpp/http-datasource/main.cpp

```diff
@@ -0,0 +1,90 @@
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <nlohmann/json.hpp>
+#include "mapget/http-datasource/datasource-server.h"
+
+using namespace mapget;
+using json = nlohmann::json;
+namespace fs = std::filesystem;
+
+// Class to encapsulate all logic of our data source
+class MyRemoteDataSource
```

Comment:
Is it a necessity to use composition for remote data sources? Why does the local one use inheritance and the remote one not?